### PR TITLE
initial contributions guide

### PR DESCRIPTION
**Pull request by ihartwig**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+* get the approval of an area lead (even though github will not enforce this)
+* if you are the area lead get the approval of another area collaborator
+* pass continuous integration tests (TravisCI)
+
+The approver will leave a comment like "ready for merge" if they approve, but they will not approve until any other commented issues are resolved. They may approve and leave a "(nit)" comment in the case of extra spaces or misspellings; the writer is responsible for resolving these before merge but does not need new approval.
+
+We use TravisCI to test that new high level code compiles. Eventually, it will run other unit tests too. Low level code is not checked by TravisCI. The approver is responsible for checking low level compilation, although the writer should check this too.
+
+The goal of our branches/pull request policy is to keep ownership of the code with the writer while controlling the quality of what is in master. It is the writer's responsiblity to sheppard their code into master. This means they should not feel bad about bothering the necessary reviewers if they have not responded in more than a day.
+
```
Comment (ihartwig) [inline CONTRIBUTING.md]: I was thinking about adding a requirement to tag relevant issues too.

**Comment by TrevorDecker** [comment]
Why do you want this in the repo instead of the wiki?

**Comment by ihartwig** [comment]
Because it is open source convention to include a CONTRIBUTING.[txt|dox|md] file if you want to restrict how people contribute to the project. See https://github.com/vim/vim/blob/master/CONTRIBUTING.md or https://gitlab.com/gitlab-org/gitlab-ce/blob/master/CONTRIBUTING.md for examples.

**Comment by TrevorDecker** [comment]
ok if thats the case then lets remove https://github.com/CMU-Robotics-Club/RoboBuggy/wiki/How-to-make-code-changes 

We should encode every thing about our policy in this doc. 
Specifically: 
* It should be explicitly stated whoever makes the pull request will be the one to merge the pull request into master.
* It should be explicitly stated whoever makes the pull request is in charge of maintaining that section of code. 
* It should be explicitly stated that using -f needs approval 
* I would also like the file be more "freshman" friendly, specifically adding the command that they need to run for each step.  


I also like the idea of adding a requirement tag policy

**Comment by ihartwig** [comment]
I think I covered most of the points you are asking for:
* line 30/line 20
* line 30/line 20
* line 20
* line 14
* I like this idea. I was too lazy at the moment.

**Pull request merged**